phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java

```diff
@@ -122,7 +123,38 @@
             ColumnFamilyDescriptorBuilder.KEEP_DELETED_CELLS,
             ColumnFamilyDescriptorBuilder.REPLICATION_SCOPE);
 
-    public static boolean areClientAndServerCompatible(long serverHBaseAndPhoenixVersion) {
+    public static class ClientServerCompatibility {
+
+        long serverVersion;
+        StringBuilder errorCode;
+        boolean isCompatible;
+
+        ClientServerCompatibility(long serverVersion) {
+            this.serverVersion = serverVersion;
+            this.errorCode = new StringBuilder();
+            this.isCompatible = true;
+        }
+
+        public int getErrorCode() {
+            int errorCode;
+            try {
+                errorCode = Integer.parseInt(this.errorCode.toString());
```

Comment:
I don't understand why we are converting this to string. Do we need to store the error code as a string for some reason?